lms/djangoapps/instructor/services.py

```diff
@@ -0,0 +1,45 @@
+"""
+Implementation of "courseware" service
+"""
+
+import logging
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+from courseware.models import StudentModule
+from instructor.views.tools import get_student_from_identifier
+import instructor.enrollment as enrollment
+
+
+log = logging.getLogger(__name__)
+
+
+class CoursewareService(object):
```

Comment:
Sorry, I don't think this is the right name for this service as it has little to do with courseware. Since this provides callbacks into instructor dashboard logic, how about "InstructorService"? Then we can register the name of the service as "instructor".